app/routes/circulars/circulars.server.ts

```diff
@@ -172,3 +173,18 @@ export async function put(subject: string, body: string, request: Request) {
     submitter: formatAuthor(user),
   })
 }
+
+export async function circularRedirect(query: string) {
+  const validCircularSearchStyles =
+    /^(?:\s*GCN\s*)?(?:\s*CIRCULAR\s*)?((\s*-?\d+(?:\.\d)?)\s*)$/i
```

Comment:
- Remove unnecessarily repeated `\s*`
- Remove unnecessary doubly-nested capturing group

```suggestion
    /^\s*(?:GCN)?\s*(?:CIRCULAR)?\s*(-?\d+(?:\.\d)?)\s*$/i
```